dotnet/src/Microsoft.Agents.AI.AGUI/AGUIChatClient.cs

```diff
@@ -199,6 +202,9 @@ public async IAsyncEnumerable<ChatResponseUpdate> GetStreamingResponseAsync(
             var threadId = ExtractTemporaryThreadId(messagesList) ??
                 ExtractThreadIdFromOptions(options) ?? $"thread_{Guid.NewGuid():N}";
 
+            // Extract state from the last message if it contains DataContent with application/json
+            JsonElement state = this.ExtractAndRemoveStateFromMessages(messagesList);
```

Comment:
why do we only supply the last message contents, not all of those? Is that because input in AG-UI allows a **single** message passing at a time?